synapse/tests/test_lib_cell.py

```diff
@@ -1254,6 +1254,15 @@ async def _fakeBackup(self, name=None, wait=True):
                             with self.raises(asyncio.TimeoutError):
                                 await asyncio.wait_for(arch, timeout=0.1)
 
+                            arch = s_t_utils.alist(proxy.iterNewBackupArchive('dupbackup', remove=True))
+                            task = core.schedCoro(arch)
+                            await asyncio.sleep(0)
```

Comment:
Is this a race condition we can eliminate via hooking or something?

Reply:
Updated with some hooks to remove the sleeps.